source/wp-content/themes/wporg-developer-2023/src/code-methods/block.php

```diff
@@ -73,42 +71,46 @@ function get_methods_content( $post_id ) {
 
 		if ( $class_methods ) {
 			usort( $class_methods, 'DevHub\compare_objects_by_name' );
-			$output .= '<ul>';
-			foreach ( $class_methods as $method ) {
-				$methods_list = '<li>';
-
-				$permalink = get_permalink( $method->ID );
-				$methods_list .= "<a href='$permalink'>";
-
-				$title = get_the_title( $method );
-				$last_colon = strrpos( $title, ':' );
-				$pos = ( false !== $last_colon ) ? $last_colon + 1 : 0;
-				$methods_list .= substr( $title, $pos );
-
-				$methods_list .= '</a>';
-
-				// Remove the filter that adds the code reference block to the content to avoid a possible infinite loop.
-				remove_filter( 'the_content', 'DevHub\filter_code_content', 4 );
-
-				$excerpt = apply_filters( 'get_the_excerpt', $method->post_excerpt, $method );
-				if ( $excerpt ) {
-					$methods_list .= ' &mdash; ' . sanitize_text_field( $excerpt );
-				}
-
-				// Re-add the filter that adds this block to the content.
-				add_filter( 'the_content', 'DevHub\filter_code_content', 4 );
-
-				if ( is_deprecated( $method->ID ) ) {
-					$methods_list .= ' &mdash; <span class="deprecated-method">' . __( 'deprecated', 'wporg' ) . '</span>';
-				}
+			$headings     = array( __( 'Name', 'wporg' ), __( 'Description', 'wporg' ) );
 
-				$methods_list .= '</li>';
+			return sprintf(
+				'<!-- wp:wporg/code-table {"id":"uses","headings":%1$s,"rows":%2$s,"itemsToShow":100,"style":{"spacing":{"margin":{"top":"var:preset|spacing|20"}}}} /--> ',
+				wp_json_encode( $headings ),
+				wp_json_encode( get_row_data( $class_methods ) )
+			);
+		}
+	}
+}
 
-				$output .= $methods_list;
-			}
-			$output .= '</ul>';
+/**
+ * Returns list of rows for the table.
+ *
+ * @param WP_Post[] $class_methods
+ * @return array[]
+ */
+function get_row_data( $class_methods ) {
+	$rows = array();
+
+	foreach ( $class_methods as $method ) {
+		// Remove the filter that adds the code reference block to the content to avoid a possible infinite loop.
+		remove_filter( 'the_content', 'DevHub\filter_code_content', 4 );
```

Comment:
When is `the_content` filter run? Since you're using the excerpt below, I don't think it is.

Reply:
If not adding that, there's an error ```[18-Jan-2024 06:00:19 UTC] PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) in /var/www/html/wp-content/plugins/query-monitor/classes/Backtrace.php on line 137``` And when I change `apply_filters( 'get_the_excerpt', $method->post_excerpt, $method )` to just `$method->post_excerpt`, everything works fine again.

I checked which functions were hooked to `get_the_excerpt`, and noticed that `wp_trim_excerpt` is one of them. This function [triggers](https://github.com/WordPress/wordpress-develop/blob/6.4/src/wp-includes/formatting.php#L3992) 'the_content' filter, leading to an infinite loop.

Reply:
TIL! Thanks for explaining that.